src/MessagePack.GeneratorCore/Generator/FormatterTemplate.tt

```diff
@@ -123,7 +123,15 @@ namespace <#= Namespace #>
                 }
             }
 
-<# if (!canOverwrite) { #>
+<# if (objInfo.HasInitOnlySetter) { #>
```

Comment:
Here, we generate initialization by object initialization expression, not by assignment or constructor argument.